js/shared.js

```diff
@@ -372,6 +372,82 @@ function undoBrowserSelection(s){
     s?.removeAllRanges?.() ?? s?.empty?.()
 }
 
+/** 
+ * Discover proper identifier for manuscript or suggest new IRI. 
+ *
+ * @async
+ * @function findShelfmark
+ * @param { string } msid Identifier to match or validate.
+ * @param { boolean } [forceNew] True to skip search for collision. *risky*
+ * @returns Valid msid string or parent IRI
+ */ 
+async function findShelfmark(msid, forceNew) {
+    try {
+        // wash msid
+        if (typeof msid !== 'string') {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add a non string.', false)
+            return
+        }
+
+        const cleanMsid = msid.replace(/[@$%*?]+/g, '') 
+        .replace(/\s+/g, ' ') 
+        .trim() 
+
+        if (cleanMsid.length === 0) {
+            const invalidInputEvent = new CustomEvent("Failed to Query Rerum. Invalid msid identifier input.")
+            globalFeedbackBlip(invalidInputEvent, 'Failed to query for Shelfmark: Attempted to add an empty string.', false)
+            return
+        }
+
+        // return washed msid or parent msid
+        if (forceNew) {
+            return cleanMsid
+        }
+        else {
+            // check for parent msid
+            const query = {
+                "body": { 
+                    "alternateTitle": { 
+                        "value": cleanMsid 
+                    } 
+                },
+                "__rerum.generatedBy" : __constants.generator
+            }
+    
+            const annotation = await fetch(`${__constants.tiny+"/query"}`, {
+                method: 'POST',
+                mode: 'cors',
+                headers: {
+                    "Content-Type": "application/json;charset=utf-8"
+                },
+                body: JSON.stringify(query)
+            })
+            .then(resp => resp.json())
+            .catch(err => {
+                console.error(err)
+                return null
+            })
+    
+            if(annotation === null){
+                const qryFail = new CustomEvent("Failed to query RERUM.")
+                globalFeedbackBlip(qryFail, 'Failed to find annotation with msid identifier.', false)
+                return
+            }
+            else if(annotations.length > 0){
+                
```

Comment:
This is a little complicated because this issue and two others bracketed the introduction of new specifications regarding Manuscript Events that impacts some higher level logic. I can talk through it with you if you keep heading this way, but it is okay to just return the `target` of the match here for now.